dfusion_rust_core/src/models/auction_element.rs

```diff
@@ -22,30 +22,37 @@ impl AuctionElement {
     }
 
     pub fn from_bytes(order_count: &mut HashMap<H160, u16>, bytes: &[u8; 113]) -> Self {
-        let account_id = H160::from(&bytes[92..112]);
+        let account_id = H160::from(&bytes[93..]);
 
-        let hopefully_null = BigEndian::read_u128(&bytes[60..76]);
+        // these go together (since sell_token_balance is emitted as u256 and treated as u128
+        let sell_token_balance = BigEndian::read_u128(&bytes[77..93]);
+        let hopefully_null = BigEndian::read_u128(&bytes[61..77]);
+        assert_eq!(hopefully_null, 0, "User has too large balance to handle.");
 
-        assert_eq!(hopefully_null, 0);
-        let sell_token_balance = BigEndian::read_u128(&bytes[76..92]);
-
-        let buy_token = u16::from_le_bytes([bytes[58], bytes[59]]);
-        let sell_token = u16::from_le_bytes([bytes[56], bytes[57]]);
+        let buy_token = u16::from_le_bytes([bytes[59], bytes[60]]);
+        let sell_token = u16::from_le_bytes([bytes[57], bytes[58]]);
         let valid_from = U256::from(u32::from_le_bytes([
-            bytes[52], bytes[53], bytes[54], bytes[55],
+            bytes[53], bytes[54], bytes[55], bytes[56],
         ]));
         let valid_until = U256::from(u32::from_le_bytes([
-            bytes[48], bytes[49], bytes[50], bytes[51],
+            bytes[49], bytes[50], bytes[51], bytes[52],
         ]));
-        let is_sell_order = bytes[47] > 0;
-        let numerator = BigEndian::read_u128(&bytes[31..47]);
-        let denominator = BigEndian::read_u128(&bytes[15..31]);
-        let remaining = BigEndian::read_u128(&bytes[0..15]);
+        let is_sell_order = bytes[48] > 0;
+        let numerator = BigEndian::read_u128(&bytes[32..48]);
+        let denominator = BigEndian::read_u128(&bytes[16..32]);
+        let remaining = BigEndian::read_u128(&bytes[0..16]);
 
+        let mut other = 0;
         let (buy_amount, sell_amount) = if is_sell_order {
-            (remaining, (numerator * remaining) / denominator)
+            if denominator > 0 {
```

Comment:
@josojo Should it be valid to put these kind of offers into the smart contract to begin with?

If so I think there is sufficient logic here that this should become a helper method. Could be called something like `compute_buy_sell_amount`

Reply:
Yes. that would make sense. I suspect that these values could be zero (as they are adjusted by subtraction in the contract).

Reply:
> Should it be valid to put this kind of offers into the smart contract, to begin with?

Currently, it is a valid input. Should it be prohibited? Well, at least I can not think of any situation, where it would make sense. But I think, it does not hurt to have the case distinction in rust anyways.

Yeah, I will put it into a separate function